edweiss-app/app/(app)/courses/[id]/forum/[postId]/index.tsx

```diff
@@ -0,0 +1,205 @@
+import RouteHeader from '@/components/core/header/RouteHeader';
+import ReactComponent, { ApplicationRoute } from '@/constants/Component';
+
+import Avatar from '@/components/Avatar';
+import For from '@/components/core/For';
+import Icon from '@/components/core/Icon';
+import TText from '@/components/core/TText';
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import RichText from '@/components/core/rich-text/RichText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import SyncStorage from '@/config/SyncStorage';
+import { CollectionOf, Document, callFunction } from '@/config/firebase';
+import t from '@/config/i18config';
+import { useCachedDynamicDocs, usePrefetchedDynamicDoc } from '@/hooks/firebase/firestore';
+import { ApplicationRouteSignature, useRouteParameters } from '@/hooks/routeParameters';
+import { useStoredState } from '@/hooks/storage';
+import { Forum } from '@/model/forum';
+import { CourseID } from '@/model/school/courses';
+import { Time } from '@/utils/time';
+import { router } from 'expo-router';
+import { Key, useMemo, useState } from 'react';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+export const PostRouteSignature: ApplicationRouteSignature<{
+	courseId: CourseID,
+	postId: Forum.PostID,
+	prefetchedPost?: Document<Forum.Post>
+}> = {
+	path: "/courses/[id]/forum/[postId]"
+}
+
+const Route: ApplicationRoute = () => {
+	const { courseId, postId, prefetchedPost } = useRouteParameters(PostRouteSignature);
+
+	const likedStorageKey = `post-liked-${postId}`;
+
+	const [post] = usePrefetchedDynamicDoc(CollectionOf<Forum.Post>(`courses/${courseId}/forum`), postId, prefetchedPost);
+	const [liked, setLiked] = useStoredState(likedStorageKey, false);
+	const initialLiked = useMemo(() => SyncStorage.get(likedStorageKey) == true, []);
+
+	const answers = useCachedDynamicDocs(`post-answers-${postId}`, CollectionOf<Forum.Answer>(`courses/${courseId}/forum/${postId}/answers`).orderBy("createdAt", "asc"));
+
+	const [answer, setAnswer] = useState("");
+
+	const [loading, setLoading] = useState(false);
+
+	const insets = useSafeAreaInsets();
+
+	if (post == undefined)
+		return <RouteHeader title='Post' />;
+
+	async function toggleLike() {
```

Comment:
Consider helping @Gutogmaia  with his upvote system, as this seems similar.

Reply:
I am, I reviewed his PR.